volatility/framework/plugins/windows/procdump.py

```diff
@@ -32,54 +34,61 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             requirements.PluginRequirement(name = 'pslist', plugin = pslist.PsList, version = (1, 0, 0)),
         ]
 
-    def _generator(self, procs):
-
+    @classmethod
+    def process_dump(cls, proc)
```

Comment:
Ideally provide documentation for what this function does in the docstring.  Also, this needs properly parameterizing (anywhere you're using `self` won't work, so things from the config pass in as something like `symbol_table` or similar.  You'll also need to pass in `context` rather than using `self._context` or `self.context`.

It would also be nice to type annotate this, but not to worry if that's problematic.  Lastly, there needs to be a colon on the end of this line...  5;)

Reply:
ah true a very nice point about `self`, but another plugin already should all that defined no? i will need to rework that part, the rest is fixed but not pushed yet

Reply:
It will, but we try not to use inheritance, and instead call class methods (hence the first parameter is cls).  This means that functionality inside a plugin doesn't suddenly change without warning, but is explicitly stored in the plugin that defined it and calls out to the plugin it needs...